modules/openfoam/src/OpenFOAM.f90

```diff
@@ -1004,42 +1005,137 @@ END SUBROUTINE CalcForceActuatorPositionsTower
 
 !--------------------------------------------------------------------------
 !> Creates the blade and tower nodes in radial and tower height co-ordinates
-SUBROUTINE OpFM_CreateActForceBladeTowerNodes(p_OpFM, ErrStat, ErrMsg)
-   TYPE(OpFM_ParameterType),INTENT(INOUT) :: p_OpFM     ! data for the OpenFOAM integration module
-   INTEGER(IntKi)                         :: ErrStat    ! Error status of the operation
-   CHARACTER(ErrMsgLen)                   :: ErrMsg     ! Error message if ErrStat /= ErrID_None
+SUBROUTINE OpFM_CreateActForceBladeTowerNodes(InitOut_AD, p_OpFM, u_OpFM, ErrStat, ErrMsg)
+   TYPE(AD_InitOutputType), INTENT(IN   )  :: InitOut_AD ! InitOut data for the OpenFOAM integration module
+   TYPE(OpFM_ParameterType),INTENT(INOUT)  :: p_OpFM     ! Parameter data for the OpenFOAM integration module
+   TYPE(OpFM_InputType),    INTENT(INOUT)  :: u_OpFM     ! Input data for the OpenFOAM integration module
+   INTEGER(IntKi)                          :: ErrStat    ! Error status of the operation
+   CHARACTER(ErrMsgLen)                    :: ErrMsg     ! Error message if ErrStat /= ErrID_None
 
    !Local variables
-   REAL(ReKi)                             :: dRforceNodes ! Uniform distance between two consecutive force nodes
-   INTEGER(IntKI)                         :: i            ! Loop variables
-   INTEGER(IntKi)                         :: ErrStat2     ! temporary Error status of the operation
-   CHARACTER(ErrMsgLen)                   :: ErrMsg2      ! temporary Error message if ErrStat /= ErrID_None
-   CHARACTER(*),   PARAMETER             :: RoutineName = 'OpFM_CreateActForceBladeTowerNodes'
+   REAL(ReKi), ALLOCATABLE                 :: cNonUniform(:)
+   REAL(ReKi), ALLOCATABLE                 :: sNonUniform(:)
+   REAL(ReKi), ALLOCATABLE                 :: pNonUniform(:)
+   REAL(ReKi), ALLOCATABLE                 :: pUniform(:)
+   REAL(ReKi), ALLOCATABLE                 :: cByS(:)
+   REAL(ReKi), ALLOCATABLE                 :: e(:)
+   REAL(ReKi)                              :: eSum, eTol
+   REAL(ReKi)                              :: bladeRoot, bladeTip, rInterp
+   INTEGER(IntKi)                          :: counter
+   REAL(ReKi)                              :: dRforceNodes ! Uniform distance between two consecutive force nodes
+   INTEGER(IntKI)                          :: i            ! Loop variables
+   INTEGER(IntKi)                          :: ErrStat2     ! temporary Error status of the operation
+   CHARACTER(ErrMsgLen)                    :: ErrMsg2      ! temporary Error message if ErrStat /= ErrID_None
+   CHARACTER(*),   PARAMETER               :: RoutineName = 'OpFM_CreateActForceBladeTowerNodes'
 
    ErrStat = ErrID_None
    ErrMsg = ""
 
    ! Line2 to Line2 mapping expects the destination mesh to be smaller than the source mesh for deformation mapping and larger than the source mesh for load mapping. This forces me to create nodes at the very ends of the blade.
```

Comment:
Is this comment still relevant?

Reply:
I think so, but I don't really know.

Reply:
We can probably drop that comment.  

If I remember correctly, we are no using a Line2 mesh and a Point mesh.  There is some kind of tolerance there on length, but I don't think it is as stringent.

Reply:
Okay, I removed the comment.